### DISCUSSION TITLE
Is memory leek?

### DISCUSSION THREAD

**Discussion by Coloryr** [AvaloniaUI/Avalonia#10819] [Q&A]
https://user-images.githubusercontent.com/26276037/228210172-741520ab-416a-470b-9b13-98954dd60320.mp4

create a new bitmap
```
Task.Run(() =>
            {
                try
                {
                    var img = ImageTemp.Load(data.Logo, cancel.Token);
                    if (img != null)
                    {
                        if (cancel.IsCancellationRequested)
                        {
                            img.Dispose();
                            GC.Collect();
                            return;
                        }
                        Dispatcher.UIThread.Post(() =>
                        {
                            Image1.Source = img;
                        });
                    }
                }
```

before create a another new bitmap
```
public void Cancel()
    {
        cancel.Cancel();

        if (Image1.Source != App.GameIcon && Image1.Source is Bitmap image)
        {
            Image1.Source = null;
            image.Dispose();
            GC.Collect();

            Image1.Source = App.GameIcon;
        }
    }
```

**Comment by Coloryr** [comment]
![6(XA0D_9NH3$TMGN1R7RZ8M](https://user-images.githubusercontent.com/26276037/228210615-78b61a09-5a7e-4258-aa4e-2312bf93252d.png)

**Comment by Coloryr** [comment]
https://github.com/Coloryr/ColorMC/blob/f651ab6476a061f46239ca9ec48409270d559843/src/ColorMC.Gui/UI/Controls/Add/FileItemControl.axaml.cs#L86
https://github.com/Coloryr/ColorMC/blob/f651ab6476a061f46239ca9ec48409270d559843/src/ColorMC.Gui/Utils/Utils.cs#L403

**Comment by Coloryr** [comment]
![RV5 IUF 6_ CU9C@G46@K5](https://user-images.githubusercontent.com/26276037/228214929-bc93fe18-1f7d-44a9-a5aa-0585390cfa89.png)

**Comment by Coloryr** [comment]
![%~1)$4VPUKSJ4A{JWPMU2SH](https://user-images.githubusercontent.com/26276037/228215042-2948e906-622f-4efb-8982-f50123be2b4c.png)

**Comment by Coloryr** [comment]
![98BQ7@VMO6ZPTUQM QL6{_J](https://user-images.githubusercontent.com/26276037/228215586-8cff5d51-861e-4974-b8b3-3d21f40c589d.png)

**Comment by timunie** [comment]
You can also bind to Tasks in Avalonia, this way you don't need to handle it that way. See: https://docs.avaloniaui.net/docs/data-binding/binding-to-tasks-and-observables

**Reply by Coloryr** [reply]
How to notify the change for task value

**Reply by Coloryr** [reply]
![{ 5 N$NG9(L8)1111SDFE_R](https://user-images.githubusercontent.com/26276037/228275877-5b743dec-c5eb-4f90-8318-1ca86abbbe0e.png)
![CFEMIGHJ C2TEQ2BK{W9LHI](https://user-images.githubusercontent.com/26276037/228275888-b13ebc5c-3da2-41bf-810a-7ad9a1c2f8d0.png)
![P7 VGT$ }O8_JV7H_L%HTDP](https://user-images.githubusercontent.com/26276037/228275914-9cc63fdd-d7bb-49b1-ae43-3f55fdfbef72.png)

**Reply by Coloryr** [reply]
ok,it work
![image](https://user-images.githubusercontent.com/26276037/228278118-88670db8-6516-4148-afbc-7b1e7e749a50.png)
![image](https://user-images.githubusercontent.com/26276037/228278172-b53e8e3e-4600-4cab-95ad-fe6529e45817.png)
![image](https://user-images.githubusercontent.com/26276037/228278208-75f1c7c8-5948-434b-a58d-ca4190f7dc0a.png)
but it will leek same.
![image](https://user-images.githubusercontent.com/26276037/228278303-3b0482f2-2e08-4cc8-b628-97e4abfbd6f6.png)

**Reply by Coloryr** [reply]
![K{I}M4{RN0Z~LT_HF62ETYU](https://user-images.githubusercontent.com/26276037/228282968-19b97298-5918-445f-83ce-ec5ac1658f0e.png)
![$ VI`PZX`U0T9I6NU_NYO4](https://user-images.githubusercontent.com/26276037/228283230-4027c902-7474-4e24-8c4e-35bd89569d80.png)

**Reply by timunie** [reply]
Also please note that each pixel is loaded into memory, so if the image has high resolution memory increases heavily as the compression which jpg or png has is not in the memory. You can use `Bitmap.DecodeToWidth(imageStream, 400))` to limit the image size to display. This has a huge impact on memory usage. 

More: https://docs.avaloniaui.net/tutorials/music-store-app/displaying-images
